src/ServicePulse.Host/app/js/views/failed_messages/controller.js

```diff
@@ -109,6 +109,22 @@
             $location.path("/failed-messages/message/" + message.id);
         };
 
+        vm.selectAllMessages = function() {
+            var selectAll = true;
+            if(vm.selectedIds.length == vm.failedMessages.length || vm.selectedIds.length > 0) {
+                selectAll = false;
+            }
+            vm.failedMessages.forEach(function(item) {
+                if (selectAll == false) {
+                    vm.selectedIds = vm.selectedIds.filter(i => i === item.id );
+                    item.selected = false;
+                } else {
+                    item.selected = true;
+                    vm.selectedIds.indexOf(item.id) >= 0 ? false : vm.selectedIds.push(item.id);
```

Comment:
if we clear the `selectedIds` collection before looping through `failedMessages`, shouldn't this ensure that `vm.selectedIds.indexOf(item.id) >= 0` is never true and we could just push without checking?